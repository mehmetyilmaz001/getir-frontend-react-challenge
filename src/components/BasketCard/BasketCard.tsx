import * as React from 'react';
import { useSelector } from 'react-redux';
import styled from 'styled-components';
import { Store } from '../../redux';
import theme from '../../style/Theme';
import Card from '../common/Card';
import BasketListItem from './components/BasketListItem/BasketListItem';
import BasketTotal from './components/BasketTotal/BasketTotal';


interface BasketCardProps {
}

const Container = styled.div`
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    gap: ${theme.spacing(4)};
`;


const TotalContainer = styled.div`
    width: 92px;
    height: 51px;
`
 

/**
 * Resuable component for displaying a basket which is connected to the redux store
 */
const BasketCard: React.FunctionComponent<BasketCardProps> = () => {
    
    const { items } = useSelector((state: Store) => state.basket);
    if(items.length === 0) {
        return <></>;
    }
    return ( 
        <Card style={{border: `solid 6px ${theme.palette.primary.main}`}}>
            <Container>
                {items.map((item, index) => <BasketListItem key={index} item={item} />)}

                <TotalContainer>
                    <BasketTotal showBasketIcon={false} theme='light'/>
               </TotalContainer>
            </Container>
        </Card>
     );
}
 
export default BasketCard;



 

 





 
