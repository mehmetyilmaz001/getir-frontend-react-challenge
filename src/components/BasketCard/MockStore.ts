import { Basket, BasketItem } from './../../types/Basket';
export const mockBasketItem: BasketItem = {
    added: 213213123,
    description: "",
    itemType: "",
    manufacturer: "",
    slug: "",
    tags: [],
    name: "test",
    price: 111,
    quantity: 1,
  };

export const mockBaskeCardStore: Basket = {
    items: [mockBasketItem],
    total: 2222
}