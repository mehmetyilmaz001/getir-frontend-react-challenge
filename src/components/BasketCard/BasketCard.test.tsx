import BasketCard from "./BasketCard";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { mockBaskeCardStore } from "./MockStore";

const mockStore = configureStore([]);

test("Basket card renders correctly", async () => {
  let store;
  let component;

  store = mockStore({
    basket: {
        ...mockBaskeCardStore
    } 
  });

  component = renderer.create(
    <Provider store={store}>
      <BasketCard/>
    </Provider>
  );
  expect(component!.toJSON()).toMatchSnapshot();
});
