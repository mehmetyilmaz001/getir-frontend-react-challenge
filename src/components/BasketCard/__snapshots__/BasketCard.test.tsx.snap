// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basket card renders correctly 1`] = `
<div>
  <div
    className="sc-bdvvtL kCZqhj"
    style={
      Object {
        "border": "solid 6px #1EA4CE",
      }
    }
  >
    <div
      className="sc-pVTFL eQkbfo"
    >
      <div
        className="sc-hKwDye OJpkq"
      >
        <div
          className="sc-eCImPb guduZA"
        >
          <h5>
            test
          </h5>
          <span
            className="sc-jRQBWg iOTVYR"
          >
            ₺111,00
          </span>
        </div>
        <div
          className="sc-gKclnd lgFLFZ"
        >
          <button
            className="sc-dkPtRN cxHOOF"
            onClick={[Function]}
          >
            <div />
          </button>
          <div
            className="sc-iCfMLu kRtLJS"
          >
            1
          </div>
          <button
            className="sc-dkPtRN cxHOOF"
            onClick={[Function]}
          >
            <div />
          </button>
        </div>
      </div>
      <div
        className="sc-jrQzAO bnErJq"
      >
        <div
          className="sc-furwcr dxZlul"
        >
          <span>
            ₺2.222,00
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
